bioblend/galaxy/jobs/__init__.py

```diff
@@ -191,6 +191,25 @@ def get_metrics(self, job_id):
         url = self._make_url(module_id=job_id) + '/metrics'
         return self._get(url=url)
 
+    def cancel_job(self, job_id):
+        """
+        creates a cancel request for a specified job
+
+        :type job_id: str
+        :param job_id: job ID
+
+        :rtype: requests.Response
+        :return: the response object.
+
+        .. note::
+         Calling ``get_state()`` to determine current job's state
+         if the state is not ok then make a delete_request
+        """
+
+        if self.get_state(job_id) != "ok":
+            url = '/'.join((self._make_url(job_id), "/discarded"))
+            return self.gi.make_delete_request(url)
```

Comment:
Can you use the hidden `_delete` method here? Not tested but I think `self._delete(id=job_id)` would be enough.

Reply:
that makes a lot more sense to use _delete, makes this bit code a bit cleaner and easier to read and resolve the issue mentioned in the prior comment about the API endpoint.